tests/core_tests.py

```diff
@@ -122,7 +122,16 @@ def test_viz_cache_key(self):
         cache_key = viz.cache_key(qobj)
 
         qobj["groupby"] = []
-        self.assertNotEqual(cache_key, viz.cache_key(qobj))
+        cache_key_with_groupby = viz.cache_key(qobj)
+        self.assertNotEqual(cache_key, cache_key_with_groupby)
+
+        self.assertNotEqual(
+            viz.cache_key(qobj), viz.cache_key(qobj, time_compare="12 weeks")
+        )
+
+        qobj["inner_from_dttm"] = "foo"
```

Comment:
Could we make this a `datetime` object for consistency?  I personally find unit tests can be a good mechanism for understanding how the code operates and thus ensuring the types are accurate helps to improve the readability.

Reply:
updated